dts/bindings/crypto/ti,cc2520-crypto.yaml

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: TI CC2520 Cryptographic Accelerator
+
+compatible: "ti,cc2520-crypto"
```

Comment:
I don't think we should add this node, we are doing this because of how we structured the driver/etc.

I think we should add a `struct device` in `ieee802154_config` that sets an optional pointer to a crypto device.  If that isn't set than `ieee802154_security_init` will fall back to a software option.

This way the CC2520 can keep the DEVICE_DEFINE() and the `struct device` will end up being local to the driver.

Reply:
fair point, just dropped CONFIG_IEEE802154_CC2520_CRYPTO_DRV_NAME for now. please check